com.lettier.movie-monad.json

```diff
@@ -20,44 +24,1331 @@
     "--talk-name=org.gnome.SettingsDaemon.MediaKeys",
     "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
     "--filesystem=host",
-    "--talk-name=ca.desrt.dconf",
-    "--env=SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt"
+    "--filesystem=xdg-run/dconf",
+    "--talk-name=ca.desrt.dconf"
   ],
+  "build-options": {
+    "strip": false,
+    "no-debuginfo": true
+  },
   "modules": [
     {
-      "name": "movie-monad",
+      "name": "haskell-platform",
+      "only-arches": [
+        "x86_64"
+      ],
+      "buildsystem": "simple",
+      "build-commands": [
+        "tar -C / -xf hp-usr-local.tar.gz",
+        "sed -i 's/\"\\$dryrun\" = \"no\" -a `id -u` -ne 0/-f \"false\"/' /usr/local/haskell/ghc-8.0.2-x86_64/bin/activate-hs",
+        "/usr/local/haskell/ghc-8.0.2-x86_64/bin/activate-hs --verbose --prefix /usr"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://haskell.org/platform/download/8.0.2/haskell-platform-8.0.2-unknown-posix--full-x86_64.tar.gz",
+          "sha256": "c68a6f278a192d5000e8feefb0e63cef1bd441d02b846029840b98a5fa3ee31a",
+          "strip-components": 0
+        }
+      ]
+    },
+    {
+      "name": "base-orphans-0.7",
+      "only-arches": [
+        "x86_64"
+      ],
+      "buildsystem": "simple",
+      "build-commands": [
+        "/usr/bin/runhaskell Setup configure --prefix=/usr",
+        "/usr/bin/runhaskell Setup build",
+        "/usr/bin/runhaskell Setup install"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "sha256": "0aaddc39e3d0bba13acfcf0009ef57bf91d2ee74b295041d63e14c6caf4dee14",
+          "url": "https://hackage.haskell.org/package/base-orphans-0.7/base-orphans-0.7.tar.gz"
+        }
+      ]
+    },
+    {
+      "name": "call-stack-0.1.0",
+      "only-arches": [
+        "x86_64"
+      ],
+      "buildsystem": "simple",
+      "build-commands": [
+        "/usr/bin/runhaskell Setup configure --prefix=/usr",
+        "/usr/bin/runhaskell Setup build",
+        "/usr/bin/runhaskell Setup install"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "sha256": "f25f5e0992a39371079cc25c2a14b5abb872fa7d868a32753aac3a258b83b1e2",
+          "url": "https://hackage.haskell.org/package/call-stack-0.1.0/call-stack-0.1.0.tar.gz"
+        }
+      ]
+    },
+    {
+      "name": "data-default-class-0.1.2.0",
+      "only-arches": [
+        "x86_64"
+      ],
+      "buildsystem": "simple",
+      "build-commands": [
+        "/usr/bin/runhaskell Setup configure --prefix=/usr",
```

Comment:
The prefix to everything should be `/app`

Reply:
Hello @TingPing,

The only app related module is the `movie-monad` module. The rest of the modules are only needed for compilation. The `movie-monad` module has the correct [prefix](https://github.com/flathub/flathub/pull/335/files#diff-acb4b5ef989eaafa8f1e7552e81c60a5R1336).

Note that I do test the manifest for correctness by building, installing, and running the Flatpak inside of a virtual machine running Fedora 27.

![screenshot_2018-04-12_19-54-47](https://user-images.githubusercontent.com/1661343/38710040-2d41ffc8-3e8c-11e8-9753-8b892d708c4a.png)

As you can see, the Flatpak runs as expected.

:+1:

Reply:
*Everything* you build goes into `/app`, `/usr` is owned by the runtime. The `cleanup` property is what then removes unused build-only files.

Reply:
As shown in the provided [example](https://github.com/flathub/org.freedesktop.Bustle/blob/master/org.freedesktop.Bustle.json#L48), the compile time resources are put in `/usr`. 

Also, can you point me to where in the documentation it says _everything_ must go into `/app`. 

It is disingenuous to pick and choose. 

:+1:

Reply:
The documentation is sadly not comprehensive yet. But they are as I said. `writable-sdk` is just a workaround as mentioned.